### PR TITLE
Add boost-gecko search box

### PR DESCRIPTION
**Pull request by ashtum**
This is a temporary solution for integrating the current search functionality from boost.org into the new website. We should explore for a more effective way of incorporating and updating the boost-gecko scripts.

The script is a build from the [new-website branch](https://github.com/ashtum/boost-gecko/tree/new-website), which has been modified to extract template parameters, such as dark/light mode and font-family, and use externally provided search buttons.

The search functionality will be further extended to include searching in guidelines, news, and mailing lists as well.

### CODE REVIEW

**Comment by ashtum** [comment]
@gregnewman, would you be able to take a look at this?

**Comment by vinniefalco** [comment]
*NOICE!*

**Comment by gregnewman** [comment]
@ashtum I'm reviewing

File: templates/includes/_header.html [hunk 1]
```diff
@@ -28,6 +28,13 @@
             <a href="{% url 'releases-most-recent' %}" class="hover:no-underline font-semibold dark:font-medium {% if request.path|slice:"0:10" == '/releases/' %}dark:text-orange text-orange dark:hover:text-white hover:text-slate {% else %}dark:text-white text-slate dark:hover:text-orange hover:text-orange {% endif %} link-none visited:text-black">Releases</a>
           </nav>
           <nav class="flex items-center space-x-3 lg:space-x-5 h-[30px]" x-data="{ 'searchOpen': false }">
+            <span class="relative">
+              <i id="gecko-search-button" data-current-boost-version="1_83_0" data-theme-mode="light" data-font-family="Cairo, sans-serif" class="inline -mt-1 cursor-pointer text-slate fas fa-search dark:text-white/50 dark:hover:text-orange hover:text-orange"></i>
```
Comment (gregnewman) [inline templates/includes/_header.html:32]: @ashtum what are the data attributes doing? Is that something boost-gecko is using?

**Review by gregnewman — APPROVED** [review]
@ashtum looks good with just the one question I had.

File: templates/includes/_header.html (hunk 1 above)
Comment (ashtum) [reply templates/includes/_header.html:32]: Yes, that's a way to pass some parameters.
There is a MutationObserver in boost-gecko to react to the changes of these attributes as well.(like toggling dark/light modes).

**Comment by ashtum** [comment]
@gregnewman Could you please merge this? I don't have the necessary access.

**Comment by 4down** [comment]
@ashtum Looks good to me, just pulled it down locally to check it out.  I'll merge!

**Pull request merged**